components/Navigation/MobileNav/MobileNav.tsx

```diff
@@ -1,60 +1,89 @@
 import Link from 'next/link';
-import { RefObject } from 'react';
+import React, { RefObject } from 'react';
+import { useRouter } from 'next/router';
 import { NavLink } from 'types/Base';
 import styles from './MobileNav.module.scss';
 
 type MobileNavItemProps = {
     link: NavLink;
+    toggleMobileMenu: (_: void) => void;
     jobsAvailable?: number;
 };
 
-const MobileNavItem = ({ jobsAvailable = 0, link }: MobileNavItemProps) => (
-    <li
-        className={`${styles.mobileNavItem} ${
-            link.title === 'Jobs' &&
-            jobsAvailable > 0 &&
-            styles.mobileNavItemWithBadge
-        }`}
-    >
-        {link.isCareersSiteInternalLink ? (
-            <Link href={link.url} scroll={false}>
-                <a className={styles.mobileNavItemLink}>
+const MobileNavItem = ({
+    jobsAvailable = 0,
+    toggleMobileMenu,
+    link,
+}: MobileNavItemProps) => {
+    const router = useRouter();
+    const redirectOrCloseMenu = (
+        event: React.MouseEvent<HTMLAnchorElement>,
+    ) => {
+        if (link.url === router.pathname) {
+            event.preventDefault();
+            toggleMobileMenu();
+        }
+    };
+    return (
+        <li
+            className={`${styles.mobileNavItem} ${
+                link.title === 'Jobs' &&
+                jobsAvailable > 0 &&
+                styles.mobileNavItemWithBadge
+            }`}
+        >
+            {link.isCareersSiteInternalLink ? (
+                <Link href={link.url} scroll={false}>
+                    <a
+                        className={styles.mobileNavItemLink}
+                        onClick={redirectOrCloseMenu}
+                    >
+                        <span className={styles.mobileNavItemTitle}>
+                            {link.title}
+                        </span>
+                        {link.title === 'Jobs' && jobsAvailable > 0 && (
+                            <span
+                                className={`${styles.mobileNavItemBadgeLink} ${styles.badge}`}
+                                aria-label={`${jobsAvailable} jobs available`}
```

Comment:
Nickpick: A "pluralize" utility function would be nice for when jobs is equal to 1.

`export const pluralize = (count: number, noun: string, nounSuffix = 's') =>
    `${count !== 1 ? nounSuffix : noun}`;`